lib/src/__tests__/DatePickerTestingLib.test.tsx

```diff
@@ -1,84 +1,116 @@
 };
 
 interface FormProps {
+  localeKey: string;
+  localizationProviderProps: Omit<LocalizationProviderProps, 'children'>;
   picker: React.FC<DatePickerProps>;
 }
 
-const Form: React.FC<FormProps> = ({ picker: Picker }) => {
-  const [date, setDate] = useState<MaterialUiPickersDate>(new Date('01/01/2020'));
-  const [locale, setLocale] = useState('en');
+const Form: React.FC<FormProps> = ({ localeKey, localizationProviderProps, picker: Picker }) => {
+  const [date, setDate] = React.useState<MaterialUiPickersDate>(new Date('01/01/2020'));
+  React.useEffect(() => {
+    moment.locale(localeKey);
+  }, [localeKey]);
   return (
-    <LocalizationProvider dateAdapter={DateFnsUtils} locale={localeMap[locale]}>
+    <LocalizationProvider {...localizationProviderProps}>
       <Picker
+        onChange={date => setDate(date)}
+        renderInput={props => <TextField {...props} />}
         value={date}
-        onChange={setDate}
         views={['year', 'month']}
-        renderInput={props => <TextField {...props} />}
       />
-      <div>
-        <Button onClick={() => setLocale('en')}>En</Button>
-        <Button onClick={() => setLocale('fr')}>Fr</Button>
-      </div>
     </LocalizationProvider>
   );
 };
 
 describe('<DatePicker />', () => {
-  it('Should add aria-invalid of true to input when date is invalid', () => {
-    render(<Form picker={DesktopDatePicker} />);
-
-    const input = screen.getByRole('textbox');
-    fireEvent.change(input, { target: { value: 'Januar 2020' } });
-
-    expect(input).toBeInvalid();
-  });
-
-  it('Should add aria-invalid of false to input when date is valid', () => {
-    render(<Form picker={DesktopDatePicker} />);
-
-    const input = screen.getByRole('textbox');
-    fireEvent.change(input, { target: { value: 'Januar 2020' } });
-    fireEvent.change(input, { target: { value: 'January 2020' } });
-
-    expect(input).toBeValid();
-  });
-
-  it('Should add aria-invalid of true to input when date is invalid for other locale', () => {
-    render(<Form picker={DesktopDatePicker} />);
-
-    const localeButton = screen.getByText('Fr');
-    const input = screen.getByRole('textbox');
-
-    fireEvent.click(localeButton);
-    fireEvent.change(input, { target: { value: 'janvie 2020' } });
-
-    expect(input).toBeInvalid();
-  });
-
-  it('Should add aria-invalid of false to input when date is valid for other local', () => {
-    render(<Form picker={DesktopDatePicker} />);
-
-    const localeButton = screen.getByText('Fr');
-    const input = screen.getByRole('textbox');
-
-    fireEvent.click(localeButton);
-    fireEvent.change(input, { target: { value: 'janvie 2020' } });
-    fireEvent.change(input, { target: { value: 'janvier 2020' } });
-
-    expect(input).toBeValid();
+  describe('locale validity', () => {
+    // iterate the different location providers and test each with each locale
+    Object.keys(dateAdapters).forEach(dateAdapterKey => {
```

Comment:
Interesting approach! I wonder what's the best approach for this. For instance, the CI is configured to run with each engine:

<img width="288" alt="Capture d’écran 2020-07-06 à 17 45 12" src="https://user-images.githubusercontent.com/3165635/86612504-808b8d00-bfb0-11ea-9884-e102adfccd89.png">

https://app.circleci.com/pipelines/github/mui-org/material-ui-pickers/1616/workflows/c55a745b-11df-4e4f-9324-9a37c8231a9e

Reply:
Is the ci capturing Hijri and Jalali?

Reply:
I don't know, but maybe it's too ambitious? I think that we can aim for smaller changes. What about we focus on adding a test case for the related issue? Also, it seems that we could already use `utilsToUse` for running the tests with different libraries.

Reply:
Sounds good.  I'll make the update.